#! /bin/sh /usr/share/dpatch/dpatch-run
## 560995_fix_ftbfs_on_sparc64.dpatch by Daniel Leidert (dale) <daniel.leidert@wgdd.de>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix an FTBFS failure on sparc64, as it tries to use SPARC32 assembly
## DP: code on SPARC64.
## DP:
## DP: <URL:http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=560995>
## DP: <URL:https://bugs.g10code.com/gnupg/issue1171>
## DP: <URL:http://lists.gnupg.org/pipermail/gnupg-commits/2009-December/008669.html>

@DPATCH@
diff -urNad trunk~/mpi/config.links trunk/mpi/config.links
--- trunk~/mpi/config.links	2008-12-11 17:39:43.000000000 +0100
+++ trunk/mpi/config.links	2010-01-22 02:17:22.404037685 +0100
@@ -124,26 +124,24 @@
 	mpi_extra_modules="udiv-qrnnd"
 	;;
     sparc64-*-linux-gnu)
-	# An extra rule because we have an report for this one only.
-	# Should be compared against the next GMP version
-	echo '/* configured for sparc64-*-linux-gnu */' >>./mpi/asm-syntax.h
-	path="sparc32v8 sparc32"
-	mpi_extra_modules="udiv"
-	;;
+       echo '/* No working assembler modules available */' >>./mpi/asm-syntax.h
+       path=""
+       ;;
     sparc64-sun-solaris2*)
-	# Got a report that udiv is missing, so we try this one
-	echo '/* configured for sparc64-sun-solaris2 */' >>./mpi/asm-syntax.h
-	path="sparc32v8 sparc32"
-	mpi_extra_modules="udiv"
-        ;;
+       echo '/* No working assembler modules available */' >>./mpi/asm-syntax.h
+       path=""
+       ;;
     sparc64-*-netbsd* | sparc64-*-freebsd* | sparc64-*-openbsd*)
        # There are no sparc64 assembler modules that work on the
        # *BSDs, so use the generic C functions.
        echo '/* No working assembler modules available */' >>./mpi/asm-syntax.h
        path=""
        ;;
+    sparc64*-*-*)
+       echo '/* No working assembler modules available */' >>./mpi/asm-syntax.h
+       path=""
+       ;;
     sparc9*-*-*     | \
-    sparc64*-*-*    | \
     ultrasparc*-*-* )
 	echo '/* configured for sparc9 or higher */' >>./mpi/asm-syntax.h
 	path="sparc32v8 sparc32"
